packages/edit-site/src/components/dataviews/view-list.js

```diff
@@ -135,38 +132,24 @@ function ViewList( {
 	paginationInfo,
 } ) {
 	const columns = useMemo( () => {
-		const _columns = [ ...fields ];
+		const _columns = [
+			...fields.map( ( field ) => {
+				const column = { ...field };
+				delete column.render;
+				column.cell = ( props ) => {
```

Comment:
The cell API is very weird and specific to tanstack, I've updated it to make it a bit clearer. I've called it "render" and it receives the "item" as a prop. This makes that function more generic and not specific to table cells. 

I believe we might want to remove `accessorFn` as well later, but didn't do it in the current PR.